### ISSUE TITLE
Add new Directory classes in IPCE to ISIS - Break up Part 1

### ISSUE THREAD

**Issue by ascbot** [DOI-USGS/ISIS3#2915]
---

Author Name: **Makayla Shepherd** (Makayla Shepherd)

Original Assignee: Makayla Shepherd

---

It appears as though the following are new classes that can be moved to ISIS. We need to verify.

New:
CloseProjectWorkOrder
CubeDnViewWorkOrder
jigsawWorkOrder
MatrixViewWorkOrder

This is just the first 4 classes on #3955

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Makayla Shepherd** (Makayla Shepherd)
Original Date: 2016-06-06T21:58:05Z

---

To Test: /work/users/mshepherd/m03993/ipce/isis

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Jesse Mapel** (Jesse Mapel)
Original Date: 2016-06-10T01:00:26Z

---

Suggested modifications from review:

* Types after @return tags need **@b** before them.  Ie. `@return bool True always` should be `@return @b bool True always`.
* In each execute()'s return documentation, True **is** WorkOrder::execute() returns true should be True **if** WorkOrder::Execute() returns true.
* Constructor documentations should be changed from: This method sets the text of the **project**, to: This method sets the text of the **work order**.

**CloseProjectWorkOrder.cpp**

* clone()'s return documentation should start with **@b CloseProjectWorkOrder** instead of **CubeDnViewWorkOrder**.

**CubeDnViewWorkOrder.cpp**

* The documentation for the project parameter in the constructor should not reference Jigsaw.
* The parameter type for isExecutable should be ImageList instead of images
* syncRedo() looks like it adds a new CubeDnView to the project's directory and then adds currentItem() to that.

**MatrixViewWorkOrder.cpp**

* The description for execute does not make sense.

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Makayla Shepherd** (Makayla Shepherd)
Original Date: 2016-06-10T17:14:42Z

---

All of the suggested changes were made with the exception of the parameter type change in CubeDNViewWorkOrder's isExecutable()

**Comment by ascbot** [comment]
---

Original Redmine Comment
Author Name: **Makayla Shepherd** (Makayla Shepherd)
Original Date: 2016-06-10T20:22:04Z

---

This ticket can be closed but it is blocked by #3956 and cannot be merged into ISIS until #3956 has been checked in
